### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by JorisCos**
About this PR

This PR introduces a wrapper around different benchmark libraries.
It makes possible to perform inference time, memory usage and FLOPs measurement on CPU.

### CODE REVIEW

File: asteroid/models/benchmarker.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import numpy as np
+
+from pypapi.exceptions import PapiNoEventError
+
+
+class Benchmarker:
+    """Wrapper around benchmark libraries on CPU. For inference time, memory
+    usage and flops measurements.
+
+        Args:
```
Comment (mpariente) [inline asteroid/models/benchmarker.py:14]: Shouldn't be indented :wink:

**Review by mpariente — COMMENTED** [review]
This looks cool !
Can you include an example code in the class docstring, with outputs as well? That would be very useful.  

@popcornell @jonashaag would you have things to add to the benchmark?

**Comment by JorisCos** [comment]
I added example but memory report is unreadable at least on github

**Review by mpariente — COMMENTED** [review]
Let's get this merged :-)
Two minor comments

**Comment by JorisCos** [comment]
Done !

**Comment by mpariente** [comment]
Great, thanks ! 

Could you add a test with a dummy model like `nn.Identity` please?

**Comment by JorisCos** [comment]
I added a test, it just runs the functions for coverage.

**Comment by mpariente** [comment]
We probably need to add the dependencies in requirements/dev.txt.

File: tests/models/benchmark_test.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class Dummy_model(torch.nn.Module):
+    def __init__(self):
+        super(Dummy_model, self).__init__()
+        self.layer = torch.nn.Identity()
+
+    def forward(self, x):
+        return self.layer(x)
+
```
Comment (mpariente) [inline tests/models/benchmark_test.py:12]:
```suggestion
```

**Comment by mpariente** [comment]
Given that there is a new profiler in torch and lightning, I wonder if it's useful to add our own here. 
Any feedback on that appreciated.

**Comment by mpariente** [comment]
Let's close this until we decide to integrate something from PyTorch. 
Thanks again for the PR.

**Pull request closed without merging**